src/Lykke.Service.FixGateway.Services/ClientOrderIdProvider.cs

```diff
@@ -58,33 +63,52 @@ private IDatabase GetDatabase()
 
         public Task<bool> CheckExistsAsync(string clientOrderId)
         {
-            return GetDatabase().SetContainsAsync(_key, clientOrderId);
+            return GetDatabase().HashExistsAsync(_key, clientOrderId);
         }
 
         public async Task RemoveCompletedAsync(Guid orderId)
         {
-            var coid = await GetClientOrderIdByOrderIdAsync(orderId);
+            var coid = await TryGetClientOrderIdByOrderIdAsync(orderId);
             await _operationsClient.Complete(orderId);
-            await GetDatabase().SetRemoveAsync(_key, coid);
+            var db = GetDatabase();
+            if (coid.hasValue)
+            {
+                await db.HashDeleteAsync(_key, coid.clientOrderId);
+            }
+
+            await db.HashDeleteAsync(_key, orderId.ToString());
         }
 
-        public async Task<string> GetClientOrderIdByOrderIdAsync(Guid orderId)
+        public async Task<(bool hasValue, string clientOrderId)> TryGetClientOrderIdByOrderIdAsync(Guid orderId)
         {
-            var context = await _operationsClient.Get(orderId);
-            var coid = JsonConvert.DeserializeObject<NewOrderContext>(context.ContextJson).ClientOrderId;
-            return coid;
+            var db = GetDatabase();
+            var cleintOrdID = await db.HashGetAsync(_key, orderId.ToString());
+            return (cleintOrdID.HasValue, cleintOrdID);
+        }
+
+        public async Task<Guid> GetOrderIdByClientOrderId(string clientOrderId)
+        {
+            var db = GetDatabase();
+            var ordId = await db.HashGetAsync(_key, clientOrderId);
+            if (!ordId.HasValue)
+            {
+                throw new InvalidOperationException($"Unknown ClientOrderID {clientOrderId}");
+            }
+            return Guid.Parse(ordId);
         }
 
         public void Start()
         {
             var operations = _operationsClient.Get(_credentials.ClientId, OperationStatus.Created).GetAwaiter().GetResult().ToArray();
-
-            GetDatabase().KeyDeleteAsync(_key).GetAwaiter().GetResult();
-            var batch = GetDatabase().CreateBatch();
+            var db = GetDatabase();
+            db.KeyDeleteAsync(_key).GetAwaiter().GetResult();
+            var batch = db.CreateBatch();
             foreach (var operation in operations)
             {
                 var coid = JsonConvert.DeserializeObject<NewOrderContext>(operation.ContextJson).ClientOrderId;
-                batch.SetAddAsync(_key, coid); // and here also
+                batch.KeyExpireAsync(_key, _keyExpirationPeriod);
+                batch.HashSetAsync(_key, coid, operation.Id.ToString()); // Do not await here
+                batch.HashSetAsync(_key, operation.Id.ToString(), coid); // Do not await here
             }
             batch.Execute();
 
```

Comment:
I would wait for batch tasks completion. 'batch.Execute()' just fires commands, but 'batch.HashSetAsync' returns related tasks.